gcc/rust/resolve/rust-default-resolver.h

```diff
@@ -0,0 +1,214 @@
+
+// Copyright (C) 2020-2023 Free Software Foundation, Inc.
+
+// This file is part of GCC.
+
+// GCC is free software; you can redistribute it and/or modify it under
+// the terms of the GNU General Public License as published by the Free
+// Software Foundation; either version 3, or (at your option) any later
+// version.
+
+// GCC is distributed in the hope that it will be useful, but WITHOUT ANY
+// WARRANTY; without even the implied warranty of MERCHANTABILITY or
+// FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+// for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with GCC; see the file COPYING3.  If not see
+// <http://www.gnu.org/licenses/>.
+
+#ifndef RUST_AST_DEFAULT_RESOLVER_H
+#define RUST_AST_DEFAULT_RESOLVER_H
+
+#include "rust-ast-visitor.h"
+#include "rust-name-resolver-2.0.h"
+
+namespace Rust {
+namespace Resolver2_0 {
+
+// FIXME: Documentation
+// does nothing for most nodes
+// apart from "scope" nodes for which it visits each member, in a name
+// resolution fashion
+class DefaultResolver : public AST::ASTVisitor
+{
+public:
+  virtual ~DefaultResolver () {} // FIXME: do we need this?
+
+  // FIXME: Do we need this?
+  // is there an easy way to overload the visitors we don't need in a derived
```

Comment:
I'm not sure I understand this comment. Could you clarify please ?

Reply:
that was me thinking while writing hoping that we could do something like `using visit(UnusedNode&) = do_nothing;` but it's stupid. I'll remove the comment and function